package.json

```diff
@@ -79,20 +79,22 @@
     "tsconfig.json"
   ],
   "scripts": {
-    "install": "(shx test -f ./script/build.js || run-s build.js) && cross-env npm_config_build_from_source=true aminya-node-gyp-build",
-    "clean": "shx rm -rf ./build ./lib/ ./prebuilds ./script/*.js ./script/*.js.map ./script/*.d.ts ./script/*.tsbuildinfo",
+    "preinstall": "run-s build.script",
+    "install": "node-gyp-build",
+    "prepare": "run-s build.library",
+    "clean": "shx rm -rf ./build ./lib/ ./prebuilds",
     "clean.temp": "shx rm -rf ./tmp && shx mkdir -p ./tmp && shx touch ./tmp/.gitkeep",
     "build.library.compat": "shx rm -rf ./lib/ts3.7 && downlevel-dts ./lib ./lib/ts3.7 --to=3.7",
     "build.library": "tsc -p ./src/tsconfig.json && run-s build.library.compat",
     "build.script": "tsc -p ./script/tsconfig.json",
     "build.js": "run-p build.script build.library",
     "build.doc": "typedoc --options ./typedoc.json && minify-all -s docs-raw -d docs --jsCompressor terser",
     "deploy.doc": "run-s build.doc && gh-pages --dist \"./docs\"",
-    "prebuild": "run-s build.js && node ./script/prebuild.js",
+    "prebuild": "node ./script/prebuild.js",
     "build.native": "node-gyp configure --release && node-gyp build --release",
     "build.native.debug": "cross-env CMAKE_BUILD_TYPE=Debug node-gyp configure --debug && cross-env CMAKE_BUILD_TYPE=Debug node-gyp build --debug",
-    "build": "run-s build.js build.native",
-    "build.debug": "run-s build.js build.native.debug",
```

Comment:
Why skipping the Js build before when it is needed?

Reply:
[The `preinstall` hook](https://docs.npmjs.com/cli/v9/using-npm/scripts#npm-install) is run before the `install` hook. That said, the scripts benefit very little from being written in TypeScript. I'd be in favor of converting them to plain JavaScript, or committing the compiled js to the repo so consumers can build the package from source *without* `typescript` installed.